### ISSUE TITLE
`no-magic-numbers`: allow numbers in expressions

### ISSUE THREAD

**Issue by Antonius-S** [eslint/eslint#13239]
Now the rule forbids usage of numbers even when they construct a constant with some expression. F.ex., if we want to correctly set a 2-min timeout value we'll write something like below getting rule report
```js
const MSEC_PER_SEC = 1000;
const SEC_PER_MIN = 60;
...
const INTRV = 2 * SEC_PER_MIN * MSEC_PER_SEC; // rule complains
```

**What rule do you want to change?**
no-magic-numbers

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
const FOO = 2;
const BAR = 3 * FOO;
```

**What does the rule currently do for this code?**
2:13 No magic number 3

**What will the rule do after it's changed?**
no message

**Are you willing to submit a pull request to implement this change?**
Probably, if the team accepts the request

**Comment by anikethsaha** [comment]
this is an intended behavior that is in the scope of this rule. 
 I don't have any strong opinion for changing the rule behavior with these cases cause this is what the rule is meant to do.

Maybe others have better approach to this.

**Comment by kaicataldo** [comment]
I agree with @anikethsaha that this is exactly what the rule is intended to enforce. In this specific example the `2` doesn't seem like a magic number because it's fairly clear what this it's referring to, but if it were a larger number it would be confusing. For instance, if the code snippet looked like this, I think the rule should warn:

```js
const MSEC_PER_SEC = 1000;
const SEC_PER_MIN = 60;
...
const INTRV = 8768652 * SEC_PER_MIN * MSEC_PER_SEC;
```

Given the above, I'm not sure how the rule would be able to distinguish when a magic number is allowed in an expression and when it isn't, and should treat all numbers in expressions as magic numbers.

**Comment by kaicataldo** [comment]
And FWIW, I do think that if you're disallowing the use of magic numbers, this should be written as:

```js
const MSEC_PER_SEC = 1000;
const SEC_PER_MIN = 60;
const NUM_OF_SEC = 2;
...
const INTRV = NUM_OF_SEC * SEC_PER_MIN * MSEC_PER_SEC;
```

**Comment by Antonius-S** [comment]
> ```js
> const MSEC_PER_SEC = 1000;
> const SEC_PER_MIN = 60;
> const NUM_OF_SEC = 2;
> ...
> const INTRV = NUM_OF_SEC * SEC_PER_MIN * MSEC_PER_SEC;
> ```

As for me, this manner looks weird and awkward.

**Comment by kaicataldo** [comment]
The team has made the difficult decision to freeze stylistic rules (please see https://eslint.org/blog/2020/05/changes-to-rules-policies for more details), and we unfortunately won't be able to accept this proposal.

Thanks for contributing to ESLint and for being willing to implement your proposal.
